### PR TITLE
[16.0][FIX] base_user_role: Improve test validation sorting values

### PR DESCRIPTION
**Pull request by victoralmau**
Improve test validation sorting values

Related to https://github.com/OCA/server-backend/pull/304

Please @pedrobaeza and @sebalix can you review it?

@Tecnativa

### CODE REVIEW

**Review by pedrobaeza — COMMENTED** [review]
I think there's no need to sort. Sets already compare equality without taking order into account.

**Comment by victoralmau** [comment]
I have proposed this change because it solves this error (installing several core modules). Sorted is something that is already being used in `other base_user_role` tests.

```
FAIL: TestUserRole.test_group_groups_into_role
Traceback (most recent call last):
  File "/opt/odoo/auto/addons/base_user_role/tests/test_user_role.py", line 272, in test_group_groups_into_role
    self.assertEqual(new_role.implied_ids.ids, user_group_ids)
AssertionError: Lists differ: [12, 8, 80, 77, 28, 82, 32, 35, 17, 19, 63, 53, 3[121 chars], 78] != [12, 8, 38, 82, 80, 32, 77, 17, 19, 63, 28, 53, 3[121 chars], 78]
First differing element 2:
80
38
```

File: base_user_role/tests/test_user_role.py [hunk 1]
```diff
@@ -269,4 +269,6 @@ def test_group_groups_into_role(self):
         new_role = self.role_model.search([("name", "=", "Test Role")])
         self.assertTrue(new_role)
         # Check that the role has the correct groups
-        self.assertEqual(new_role.implied_ids.ids, user_group_ids)
+        self.assertEqual(
+            sorted(set(new_role.implied_ids.ids)), sorted(set(user_group_ids))
+        )
```
Comment (pedrobaeza) [inline base_user_role/tests/test_user_role.py]: I mean, please try this:

```suggestion
        self.assertEqual(set(new_role.implied_ids.ids), set(user_group_ids))
```

Comment (victoralmau) [reply base_user_role/tests/test_user_role.py]: It also works!

**Review by pedrobaeza — APPROVED** [review]

**Review by CarlosRoca13 — APPROVED** [review]

**Pull request merged**